### PR TITLE
Updates for new WCSim versions with proper tracking of boundary crossings

### PR DESCRIPTION
**Pull request by nickwp**
New WCSim has proper tracking of particles crossing bounaries; in particular crossing the ID boundary can be used to exclude non-fully contained events. Tested with WCTE events from latest WCSim.

Also includes some other minor fixes / improvements

### CODE REVIEW

**Review by kmtsui — APPROVED** [review]
I don't know about the h5 conversion, otherwise the boundary points operations are good

**Review by pdeperio — APPROVED** [review]
quick review and in-person discussion with Ka Ming to confirm compatibility with IWCD and entering events as well, and no speed loss.

File: root_utils/event_dump.py [hunk 1]
```diff
@@ -101,6 +109,8 @@ def dump_file(infile, outfile):
         event_id[ev] = ev
         root_file[ev] = infile
 
+        print(ev)
```
Comment (kmtsui) [inline root_utils/event_dump.py:112]: oh forgot to remove this printout?

Comment (pdeperio) [reply root_utils/event_dump.py:112]: [removed](https://github.com/WatChMaL/DataTools/commit/4ea8d925fdce19eaa06f807d418a87fb67a30133)

**Pull request merged**